test/params_conversion_test.c

```diff
@@ -9,13 +9,19 @@
  */
 
 #include <string.h>
-#include <inttypes.h>
 #include <openssl/params.h>
 #include "testutil.h"
 
-#ifdef OPENSSL_SYS_WINDOWS
-# define strcasecmp _stricmp
-#endif
+/* On machines that dont support <inttypes.h> just disable the tests */
+# if !defined(PRId8)
```

Comment:
Since strtoimax, i.e the availability of intmax_t is the problem, then I would suggest that the comment should say so, and that it would be more intuitive to check if `INTMAX_MAX` is defined.

Reply:
No that comes from stdint.h

Reply:
There seems to be no appropriate @define inside inttypes.h

Reply:
and e_os2 has paths that just include stdint.h if inttyped.h is not included

Reply:
So the comment is correct

Reply:
We could add another #define when inttype.h is included in e_os2. Not sure thats much better

Reply:
On most platforms that *I* know of, inttypes.h includes stdint.h if the latter is available...  however, what you say suggests that we should include inttypes.h more widely and not just on platforms we know to not have stdint.h.

Reply:
(so the logic would be reversed, that we only include stdint.h where we know that inttypes.h isn't available)

Reply:
Err thats what it does now?

# if defined(OPENSSL_SYS_UEFI)
typedef INT8 int8_t;
typedef UINT8 uint8_t;
typedef INT16 int16_t;
typedef UINT16 uint16_t;
typedef INT32 int32_t;
typedef UINT32 uint32_t;
typedef INT64 int64_t;
typedef UINT64 uint64_t;
# elif (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L) || \
     defined(__osf__) || defined(__sgi) || defined(__hpux) || \
     defined(OPENSSL_SYS_VMS) || defined (__OpenBSD__)
#  include <inttypes.h>
# elif defined(_MSC_VER) && _MSC_VER<=1500
/*
 * minimally required typdefs for systems not supporting inttypes.h or
 * stdint.h: currently just older VC++
 */
typedef signed char int8_t;
typedef unsigned char uint8_t;
typedef short int16_t;
typedef unsigned short uint16_t;
typedef int int32_t;
typedef unsigned int uint32_t;
typedef __int64 int64_t;
typedef unsigned __int64 uint64_t;
# else
#  include <stdint.h>
# endif

Reply:
In both the stdint and inttype paths INTMAX_MAX will be defined.
I need to disable the tests only on the inttype path (which contains the extra functions)

Reply:
I gotta say this has my head spinning for the moment...  I'll get back on this later.

Reply:
@ping

Reply:
@levitte just to recap...
inttypes.h contains the definition of strtoimax
inttypes.h includes stdint.h (and stdint.h is the one that includes INTMAX_MAX)
So the testing of INTMAX_MAX does not mean that strtoimax exists because there are
#ifdef blocks inside e_os2.h that include EITHER inttypes OR stdint.h.
Includig stdint.h will give you INTMAX_MAX but not strtoimax.

Reply:
ping

Reply:
ping @openssl/omc

Reply:
Revisiting this, I wonder where the definition of `PRId8` would come from.  It should supposedly come from `inttypes.h` (or so tells me https://en.cppreference.com/w/c/types/integer), but since that's not included by the file, I can't see how it's ever going to be defined.

Since `strtoimax` seems to be the problem (and potentially the related types `intmax_t` and `uintmax_t`), would it be possible to alias it to something sensible?

Reply:
I will try again, but I dont think it worked when I tried just changing them in the params_test.